lib/std/json/static.zig

```diff
@@ -34,6 +34,17 @@ pub const ParseOptions = struct {
     /// The default for `parseFromTokenSource` with a `*std.json.Reader` is `std.json.default_max_value_len`.
     /// Ignored for `parseFromValue` and `parseFromValueLeaky`.
     max_value_len: ?usize = null,
+
+    /// If false, calling any parseFromSlice* or parseFromTokenSource* functions
+    /// where the given json object is only partially parsed is an assertion failure.
+    /// Setting this to true allows parseFrom* functions to parse only part of a json.
+    allow_partial: bool = false,
+
+    /// This determines whether parsed dynamic elements such as strings should always trigger an allocation,
+    /// or if a reference to the given buffer should be used.
+    /// If the reader or scanner's internal buffer won't outlive the result of the `parseFrom*` function,
+    /// use .alloc_always to ensure dynamic elements are duplicated into new buffers.
+    allocate: AllocWhen = .alloc_if_needed,
```

Comment:
I suggest this change to the optional behavior.
```suggestion
    /// This determines whether strings should always be copied,
    /// or if a reference to the given buffer should be preferred if possible.
    /// The default for `parseFromSlice` or `parseFromTokenSource` with a `*std.json.Scanner` input is `.alloc_if_needed`.
    /// The default with a `*std.json.Reader` input is `.alloc_always`.
    /// Ignored for `parseFromValue` and `parseFromValueLeaky`.
    allocate: ?AllocWhen = null,
```

Reply:
I'm working on implementing this change. I'll note here that this PR is now technically a change in the default behavior, though I agree that it's for the better.